src/Bridge/Symfony/Bundle/Action/SwaggerUiAction.php

```diff
@@ -63,29 +63,33 @@ public function __invoke(Request $request)
      *
      * @return array
      */
-    private function getContext(Request $request): array
+    private function getContext(Request $request, Documentation $documentation): array
     {
         $context = [
             'title' => $this->title,
             'description' => $this->description,
             'formats' => $this->formats,
-            'shortName' => null,
-            'operationId' => null,
         ];
 
-        if (!$request->isMethodSafe(false) || null === $resourceClass = $request->attributes->get('_api_resource_class')) {
-            return $context;
-        }
+        $swaggerData = [
+            'url' => $this->urlGenerator->generate('api_doc', ['format' => 'json']),
+            'spec' => json_decode($this->serializer->serialize($documentation, 'json')),
```

Comment:
Maybe can you try `$this->serializer->normalize` instead (to avoid the double decoding).